test_conformance/conversions/conversions_data_info.h

```diff
@@ -608,18 +661,30 @@ void DataInfoSpec<InType, OutType>::conv_sat(OutType *out, InType *in)
     }
 }
 
-template <typename InType, typename OutType>
-void DataInfoSpec<InType, OutType>::init(const cl_uint &job_id,
-                                         const cl_uint &thread_id)
+template <typename InType, typename OutType, bool InFP, bool OutFP>
+void DataInfoSpec<InType, OutType, InFP, OutFP>::init(const cl_uint &job_id,
+                                                      const cl_uint &thread_id)
 {
     uint64_t ulStart = start;
     void *pIn = (char *)gIn + job_id * size * gTypeSizes[inType];
 
-    if (std::is_integral<InType>::value)
+    if (is_in_half())
+    {
+        const unsigned sclamp =
+            std::is_signed<OutType>::value ? 0xffff : 0x7fff;
+
+        InType *o = (InType *)pIn;
+
+        for (unsigned i = start; i < size; i++)
+        {
+            o[i] = static_cast<InType>(i % sclamp);
+        }
+    }
```

Comment:
This whole section of code doesn't look right and doesn't follow the same pattern as for floats and doubles:

* Why the static cast to an `InType`?  This might be OK if it's trying to cast to a `cl_ushort` to effectively do a bitcast, but it's confusing at the very least.
* Why the modulo by `sclamp`?  The code for floats and doubles only has fixup code if we aren't saturating:

```c++
        if (kUnsaturated == sat)
        {
            InType *f = (InType *)pIn;
            for (i = 0; i < size; i++) f[i] = clamp(f[i]);
        }
```

Reply:
Please notice my initial description for the PR above (_Please take a closer look at method DataInfoSpec<>::init under condition is_in_half()..._). The code you refer to is an result of searching a way to pass the test with `fp16` implementation I am working with (`11th Gen Intel(R) Core(TM) i7-11800H`). Approach of generating input values similar to `double`s and `float`s failed in couple of conversion cases. I will appreciate your expertise here. Should we test ALL possible values from cl_half (ushort) scope ?

Reply:
Done, I have adapted this piece of code to follow the same pattern as for floats and doubles.

The obstacle to do so earlier was unclear idea behind `specialValuesFloat`/`specialValuesDouble` (I have mentioned in the description of this PR). I just borrowed such values for `specialValuesHalf` from `bruteforce` test.